runtime/domain.c

```diff
@@ -1096,8 +1096,10 @@ static void* domain_thread_func(void* v)
     caml_gc_log("Domain starting (unique_id = %"ARCH_INTNAT_PRINTF_FORMAT"u)",
                 domain_self->interruptor.unique_id);
     CAML_EV_LIFECYCLE(EV_DOMAIN_SPAWN, getpid());
+    /* FIXME: ignoring errors and asynchronous exceptions during
+       domain initialization is unsafe and/or can deadlock. */
     caml_domain_initialize_hook();
-    caml_callback(ml_values->callback, Val_unit);
+    caml_callback_exn(ml_values->callback, Val_unit);
```

Comment:
Is the idea here that `domain_thread_func` is used as the `start_routine` for `pthread_create` and hence, it is not sensible for it to throw an exception? The `Domain.spawn` function catches all the synchronous exceptions. Are we  concerned with asynchronous exceptions here?

Reply:
More work is to be done to correct handle exceptions here. In this case, an asynchronous exception can appear indeed. As discussed elsewhere, this is for another PR.

Reply:
Ok.